compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/CompilerPhaseRunner.java

```diff
@@ -233,8 +233,9 @@ private boolean stopCompilation(BLangPackage pkgNode, CompilerPhase nextPhase) {
 
     private boolean checkNextPhase(CompilerPhase nextPhase) {
         return (!isToolingCompilation && nextPhase == CompilerPhase.CODE_ANALYZE) ||
-                nextPhase == CompilerPhase.COMPILER_PLUGIN ||
-                nextPhase == CompilerPhase.DESUGAR;
+                nextPhase == CompilerPhase.COMPILER_PLUGIN || nextPhase == CompilerPhase.DESUGAR ||
+                nextPhase == CompilerPhase.BIR_GEN;
```

Comment:
I added this to prevent an NPE in BIROptimizer. We can remove it after handling closures properly.

Reply:
Create an issue and point it here.

Reply:
Created issue https://github.com/ballerina-platform/ballerina-lang/issues/34396 to revert this

Reply:
let's add this and other improvements suggested in https://github.com/ballerina-platform/ballerina-lang/pull/32697 as a separate PR.